src/formaction.cpp

```diff
@@ -351,6 +372,12 @@ void FormAction::handle_cmdline(const std::string& cmdline)
 			}
 		} else if (cmd == "dumpform") {
 			v->dump_current_form();
+		} else if (cmd == "exec") {
+			if (tokens.size() != 1) {
+				v->show_error(_("usage: exec <cmd>"));
+			} else {
+				process_op(v->get_keymap()->get_opcode(tokens[0]));
```

Comment:
`process_op` returns `bool` indicating if the operation got executed; please check it and show an error if it's `false`. This should help with e.g. the user trying to execute a non-existent operation.

Reply:
It seems unwinding the callstack of `process_op`, if no command is found in `ListFormAction::process_operation`, `true` is returned anyways.

`op == OP_NIL` seems to be the way to go